shared/js/background/components/dnr.js

```diff
@@ -0,0 +1,59 @@
+import browser from 'webextension-polyfill'
+import ATB from '../atb'
+import { flushSessionRules } from '../dnr-session-rule-id'
+import { clearInvalidDynamicRules } from '../dnr-utils'
+import { refreshUserAllowlistRules } from '../dnr-user-allowlist'
+import { ensureGPCHeaderRule } from '../dnr-gpc'
+import { onConfigUpdate } from '../dnr-config-rulesets'
+
+/**
+ * @typedef {import('./tds.js').default} TDS
+ * @typedef {import('../settings.js')} Settings
+ * @typedef {import('./trackers.js').default} Trackers
+ */
+
```

Comment:
I think with the name being "DNR" it's not super clear what the component is for, I could imagine trying to explain to someone in a code review why putting the DNR logic for a new privacy feature in here is a bad idea, and them arguing "well it's called DNR, so why not?".

Not sure what you think, and I'm not sure what a better name would be. Perhaps a good option could be to add a nice comment to explain what the DNR component is for? I figure if we explain to folks it's just for instrumenting things like updating the rulesets on install/update, rather than to contain the feature-specific logic, it might be enough. WDYT?

Reply:
Good point. I renamed the class and added a comment to describe what it does.